test/unit/proxy_test.rb

```diff
@@ -483,16 +483,6 @@ def test_apicast_configuration_driven
                               api_backend: 'http://example.com',
                               api_test_path: '/path',
                               apicast_configuration_driven: false)
-    ::ApicastV1DeploymentService.any_instance.expects(:deploy).with(proxy).returns(true)
-
-    analytics.expects(:track).with('Sandbox Proxy Deploy', success: true)
-    analytics.expects(:track).with('Sandbox Proxy updated',
-                                   api_backend: 'http://example.com:80',
-                                   api_test_path: '/path',
-                                   success: true)
-    analytics.expects(:track).with('APIcast Hosted Version Change', {enabled: false, service_id: proxy.service_id, deployment_option: 'hosted'})
-    Logic::RollingUpdates.stubs(skipped?: true)
-
```

Comment:
This should not be removed, it's not related to the `v1_compatible` option. Instead, you should probably extract a new test:
`save_and_deploy` tests the method save_and_deploy
`save_and_deploy_v1_regression` tests that it does not throw an uncontrolled error or something like that

Reply:
Agree with @josemigallas just keep the test and remove (or change) it when ApicastV1DeploymentService is removed. We will see when the time comes